python/lsst/meas/base/dataIds.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+# Copyright 2008-2015 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+import collections
+
+import lsst.pex.logging
+import lsst.pex.exceptions
+import lsst.afw.table
+import lsst.afw.image
+import lsst.coadd.utils
+from lsst.geom import convexHull
+
+class PerTractCcdDataIdContainer(lsst.coadd.utils.CoaddDataIdContainer):
```

Comment:
This applies to CCDs, but inherits from `CoaddDataIdContainer`. That seems confusing -- is it necessary?

In fact, I think all you're pulling from `Coadd..` is the `getSkymap()` method. That's really just memoizing `namespace.butler.get(datasetType + "_skyMap")`, and you're also doing that memoization directly (`if skymap is None:`, etc, at line 86 below).